packages/crayons-core/src/components/icon/icon.tsx

```diff
@@ -1,51 +1,155 @@
   @Prop() name: string;
-
   /**
-   * Size of the icon, specified in number of  pixels.
-   * Default value is 12px defined using the --icon-size css variable.
+   * Size of the icon, specified in number of  pixels. This will be square coordinates of (w X h) = size X size
    */
-  @Prop() size: number;
-
+  @Prop() size = 16;
+  /**
+   * Root Margin in px or percentage for intersection observer
+   */
+  @Prop() xRootMargin = '50px';
+  /**
+   * Width of the icon, specified in number of  pixels.
+   */
+  @Prop() width = undefined;
+  /**
+   * Height of the icon, specified in number of  pixels.
+   */
+  @Prop() height = undefined;
   /**
    * Color in which the icon is displayed, specified as a standard CSS color or as a HEX code.
    */
   @Prop() color = '';
+  /**
+   * Enable Intersection Observer. Default is true.
+   */
+  @Prop() lazy = 'true';
+  /**
+   * Name of External Library to be used
+   */
+  @Prop() library = 'crayons';
+
+  @Element() el: HTMLElement;
+
+  @State() private visible = false;
+
+  private intersectionObserver: IntersectionObserver;
 
   @State() svgHTML = '';
 
-  @Watch('name')
-  private setSVGState(iconName: string) {
-    this.svgHTML = icons[iconName];
+  @State() ico_height = undefined;
+
+  @State() ico_width = undefined;
+
+  @State() eval_xOb = false;
+
+  @Watch('name') private async setSVGState(): Promise<void> {
+    const { name, visible, lazy, library } = this;
+    if (lazy === 'false') this.eval_xOb = true;
+    else if (lazy === 'true' || lazy === undefined) {
+      this.eval_xOb = visible;
+    }
+    if (!Build.isBrowser || !name || !this.eval_xOb) {
+      return;
+    }
+
+    if (
+      name.indexOf('svg') > -1 &&
+      name.indexOf('viewBox') > -1 &&
+      name.indexOf('fill') > -1
+    )
+      this.svgHTML = this.name;
+    else {
+      this.svgHTML = await fetchIcon(name, library);
+    }
+    this.ico_height = this.height !== undefined ? this.height : this.size;
+    this.ico_width = this.width !== undefined ? this.width : this.size;
   }
 
-  componentWillLoad() {
-    this.setSVGState(this.name);
+  connectedCallback(): void {
+    this.waitUntilVisible(() => {
+      this.visible = true;
+      this.setSVGState();
+    });
+  }
+
+  disconnectedCallback(): void {
+    if (this.intersectionObserver) {
+      this.intersectionObserver.disconnect();
+      this.intersectionObserver = undefined;
+    }
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.setSVGState();
   }
 
   render() {
-    const style = {
-      '--icon-color': this.color,
-    };
+    return (
+      <div
+        class='icon'
+        style={{
+          '--icon-color': `${this.color}`,
+          'height': ` ${this.ico_height}px`,
+          'width': `${this.ico_width}px`,
+        }}
+        innerHTML={this.svgHTML}
+      ></div>
+    );
+  }
 
-    if (this.size !== undefined) {
-      style['--icon-size'] = `${this.size}px`;
+  private waitUntilVisible(callback: () => void): void {
```

Comment:
Can we move the intersection observer browser check into a util?

Reply:
This is very small code with no chances of failing.It cannot be part of separate utils as its a sort of useScreen hook equivalent of react. I have optimized it further. If we don't need it , just pass <fw-icon name="xyz" nolazy />. 
Tested and closing the coment.